### DISCUSSION TITLE
Controller : type mismatch

### DISCUSSION THREAD

**Discussion by ootsun** [react-hook-form/react-hook-form#7950] [Q&A]
Nothing is working...
Do you know where this might come from?
react-hook-form: 7.26.1

`import {Controller, useForm} from 'react-hook-form';`

`const {register, handleSubmit, formState, reset, setValue, getValues, control} = useForm({
    mode: 'onTouched'
  });`
![image](https://user-images.githubusercontent.com/25452154/156418930-0055d573-4fa3-4462-9533-283a4bcaf6d0.png)

**Comment by ootsun** [comment]
This console.log prints undefined : 


![image](https://user-images.githubusercontent.com/25452154/156430933-77ea68bc-f448-4a6d-993e-3583606edfe8.png)
![image](https://user-images.githubusercontent.com/25452154/156430707-1c21c03f-cb2a-4f24-a4c8-369ff4d51d76.png)

**Reply by bluebill1049** [reply]
`field: { onChange }`

**Reply by ootsun** [reply]
Sorry, but what do you mean by that?

**Reply by bluebill1049** [reply]
Refer to the doc you are deconstruct incorrect from the render props

**Reply by ootsun** [reply]
Indeed, it works, thanks! The warning are still present, but I guess they aren't pertinent.
